src/controllers/neighborhoods.ts

```diff
@@ -37,4 +38,36 @@ neighborhoodsRouter.put('/:id', catchError(async (req, res) => {
   res.status(200).send(`Neighborhood '${updatedNeighborhood.name}' has been updated.`);
 }));
 
+neighborhoodsRouter.post('/', middleware.userExtractor, catchError(async (req: CustomRequest, res: Response) => {
+  const userId: number = req.user?.id as number; // user.id shoul be extracted from the middleware
+  req.body.admin_id = userId; // adding user_id as admin_id to request.body
+  const createNeighborhoodData = await routeHelpers.generateCreateNeighborhoodData(req.body);
+
+  const newNeighborhood: Neighborhood = await prismaClient.neighborhood
+    .create({ data: createNeighborhoodData });
+
+  await routeHelpers.connectUsertoNeighborhood(userId, newNeighborhood.id);
+
+  const newNeighborhoodWithRelatedFields = await routeHelpers
+    .generateNeighborhoodDataWithRelatedFields(newNeighborhood.id);
+
+  res.status(201).json(newNeighborhoodWithRelatedFields);
+}));
+
+neighborhoodsRouter.post('/:id/join', middleware.userExtractor, catchError(async (req: CustomRequest, res: Response) => {
+  const userId = req.user?.id as number; // user should be extracted by the middleware
+  const neighborhoodId = Number(req.params.id);
+
+  if (!neighborhoodId || Number.isNaN(neighborhoodId)) {
+    res.status(400).send({ error: 'Unable to parse URL' });
+  } else {
+    await routeHelpers.connectUsertoNeighborhood(userId, neighborhoodId);
+
+    const neighborhoodWithUsers = await
+    routeHelpers.generateNeighborhoodDataWithRelatedFields(neighborhoodId);
+
+    res.status(200).send(neighborhoodWithUsers);
```

Comment:
I'm not sure it is necessary to send back the neighborhood. Maybe a message such as `You successfully joined NEIGHBORHOOD_NAME` would be more appropriate. What do you think?

Reply:
Yes.

I just realized that for React frontends, it makes no sense for the backend API to send detailed data. 

The data to display at the frontend (aka the state) is managed by React itself.

Reply:
Made the change.